glimpse/encoders/cnn.py

```diff
@@ -0,0 +1,158 @@
+from glimpse.neural_net import NeuralNet
+import tensorflow as tf
+import numpy as np
+from helpers.utils import load_pickle
+from helpers.definitions import data_dir
+
+
+class CNN(NeuralNet):
+  
+  def __init__(self, n_conv_layers=0, n_fcl_layers=0):
+    NeuralNet.__init__(self)
+    
+    self.n_conv_layers = n_conv_layers
+    self.n_fcl_layers = n_fcl_layers
+    
+    # Init empty attrs
+    self.X_train = None
+    self.Y_train = None
+    self.X_val = None
+    self.Y_val = None
+    self.X_test = None
+    self.Y_test = None
+
+    # Extract pickled train, validation, and test data
+    self.extract_data()
+    
+    # Define non-static params left out of params.json
+    self.define_dynamic_params()
+    
+    # Establish weight placeholders
+    self.x = tf.placeholder(shape=[None, self.params.W, self.params.H, self.params.C], dtype=tf.float32)
+    self.y = tf.placeholder(shape=[None], dtype=tf.int32)  # wrong datatype --> currently dealing with dtype='|S8749'
+    self.o = self.x
+    
+    # Construct the network
+    self.build_nn()
+    
+  def extract_data(self):
+    for set in ['train', 'val', 'test']:
+      print 'Unpickling {} set'.format(set)
+      data = load_pickle('{}/{}.pkl'.format(data_dir, set))
+      
+      x, y = data['data'], data['labels']
+      
+      setattr(self, 'X_{}'.format(set), x)
+      setattr(self, 'Y_{}'.format(set), y)
+    
+    print 'X_train.shape = {}, Y_train.shape = {}'.format(self.X_train.shape, self.Y_train.shape)
+    
+  def define_dynamic_params(self):
+    num_train_images = self.X_train.shape[0]  # => 6 right now
```

Comment:
Using only 10 images right now just to get the hang of things in a 6/2/2 breakdown among train/val/test